pkg/build/generator/generator.go

```diff
@@ -0,0 +1,243 @@
+package generator
+
+import (
+	"fmt"
+
+	"github.com/golang/glog"
+
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api/errors"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+	buildapi "github.com/openshift/origin/pkg/build/api"
+	imageapi "github.com/openshift/origin/pkg/image/api"
+)
+
+// BuildGenerator is a central place responsible for generating new Build objects
+// from BuildConfigs and other Builds.
+type BuildGenerator struct {
+	Client GeneratorClient
+}
+
+type GeneratorClient interface {
+	GetBuildConfig(ctx kapi.Context, name string) (*buildapi.BuildConfig, error)
```

Comment:
Methods defined in BuildConfigGetter, BuildConfigUpdater are reaching REST API endpoints, whereas above are reaching straight to etcd, see [here](https://github.com/openshift/origin/pull/1275/files#diff-3999d64b0b8fa9423b47f9feb61db2b0R216)